apps/aehttpserver/lib/aehttpserver/web/controllers/new_tx_controller.ex

```diff
@@ -5,10 +5,14 @@ defmodule Aehttpserver.Web.NewTxController do
   alias Aecore.Txs.Pool.Worker, as: Pool
 
   def new_tx(conn, _params) do
-    conn.body_params
-    |> Serialization.tx(:deserialize)
-    |> Pool.add_transaction()
+    deserialized_tx = Serialization.tx(conn.body_params, :deserialize)
 
-    json(conn, %{:status => :new_tx_added})
+    case Pool.add_transaction(deserialized_tx) do
+      :error ->
```

Comment:
Shouldn't we have the successful case first here?

Reply:
Does it matter?

Reply:
That's just the flow of reading the code, usually we have the success case first unless we have to pattern match the error before that.